parser-typechecker/src/Unison/Codebase/Editor/Propagate.hs

```diff
@@ -204,14 +233,20 @@ propagate patch b = case validatePatch patch of
             seen' = seen <> Set.fromList (view _1 . view _2 <$> joinedStuff)
             writeTypes =
               traverse_ (\(Reference.DerivedId id, tp) -> eval $ PutDecl id tp)
-          writeTypes (Map.elems componentMap)
```

Comment:
This was wrong. `componentMap` has the _old types_, so we were actually just writing those to the codebase.